lib/centurion/deploy_dsl.rb

```diff
@@ -85,23 +69,38 @@ def health_check(method)
    set(:health_check, method)
   end
 
+  def defined_service
+    service = fetch(:service, {})
```

Comment:
This looks like it returns an instance of `Service` if that's already defined previously with `set(:service, ...)`, and otherwise returns a hash. That breaks my expectations of what the method does. I believe it should always return a `Service`.